packages/workbox-window/src/Workbox.ts

```diff
@@ -148,6 +145,8 @@ class Workbox extends WorkboxEventTarget {
 
     // If an "own" SW is already set, resolve the deferred.
     if (this._sw) {
+      this._sw.addEventListener(
+          'statechange', this._onStateChange, {once: true});
```

Comment:
You can remove the `{once: true}` here, as that only applied in the case where the next expected event was `redundant`. And since no lifecycle events are ever going to fire after `redundant` anyway, I think it's OK to just let that be the mechanism for stopping future events.